core/emulate.c

```diff
@@ -382,6 +384,18 @@ static em_status_t segmented_write(struct em_context_t *ctxt,
     return ctxt->ops->write_memory(ctxt->vcpu, la, data, size, flags);
 }
 
+static void parse_bit_operands(struct em_context_t *ctxt)
+{
+    if (ctxt->dst.type == OP_MEM && ctxt->src1.type == OP_REG) {
+        int64_t offset;
+
+        offset = READ_GPR(ctxt->src1.reg.index, ctxt->src1.size);
```

Comment:
Fixed at dcf0a41.

For now, I'm just post-processing the offset by casting the value to the corresponding signed type (based on the operand size). Same approach is done by other hypervisors. If we see more instructions that have similar behavior, we can make a helper function instead.

Note that these casts occur only when `ctxt->dst.type == OP_MEM && ctxt->src1.type == OP_REG`. But that's alright since any other situation will mask the bit offset, so signedness can be ignored.